### PR TITLE
Remove Subject Alternate Name Sanitization in Workload Cert Creation

### PR DESCRIPTION
**Pull request by nimanch**
1.  Remove Subject Alternate Name Sanitization of Common Name, this leads to TLS Handshake Errors since the hostname differs from the SAN. 
2. Added Unit Test
3. Enable 1ES Agents on CI E2E Tests

End-End CI Run : https://msazure.visualstudio.com/One/_build/results?buildId=48717636&view=logs&j=f1061aa8-bee6-5103-5b5c-2b2c91b98c58

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Review by damonbarry — COMMENTED** [review]
Pipeline stuff looks good. I'd recommend that you get an area owner to look at the changes you made to the workload endpoint (Rust).

File: edgelet/edgelet-http-workload/src/module/cert/server.rs [hunk 1]
```diff
@@ -133,12 +131,25 @@ fn server_cert_extensions(
...
 #[cfg(test)]
 mod tests {
+    use crate::module::cert::CertificateResponse;
     use http_common::server::Route;
 
     use edgelet_test_utils::{test_route_err, test_route_ok};
 
     const TEST_PATH: &str = "/modules/testModule/genid/1/certificate/server";
 
+    const MODULENAME: &str = "testModule";
```
Comment (gordonwang0) [inline edgelet/edgelet-http-workload/src/module/cert/server.rs:141]: Nit: should be MODULE_NAME

Also, replace all the other instances of "testModule" with this.

File: edgelet/edgelet-http-workload/src/module/cert/server.rs [hunk 2]
```diff
@@ -95,8 +95,6 @@ where
         let common_name_san = if std::net::IpAddr::from_str(&common_name_san).is_ok() {
             super::SubjectAltName::Ip(common_name_san)
         } else {
-            let common_name_san = super::sanitize_dns_name(common_name_san);
```
Comment (gordonwang0) [inline edgelet/edgelet-http-workload/src/module/cert/server.rs:98]: Remove sanitize_dns_name from identity certs too, then delete the function.

**Review by gordonwang0 — CHANGES_REQUESTED** [review]

File: edgelet/edgelet-http-workload/src/module/cert/server.rs (hunk 1 above)
Comment (nimanch) [reply edgelet/edgelet-http-workload/src/module/cert/server.rs]: done

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**